### PR TITLE
Create disable-alerts-health-management-api.md

### PR DESCRIPTION
**Pull request by sashwathn**
For the "How to mute alerts during maintenance windows or scheduled backups?" blog post. It is more or less a recall to our Learn document.

### CODE REVIEW

File: blog/disable-alerts-health-management-api.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+keywords: [netdata,alerts,silence,disable,mute,maintenance window]
+authors: satya
+---
+
+For all SREs, it is absolutely crucial to filter out expected events during maintenance windows and quickly pinpoint critical issues in your infrastructure. Every minute is crucial while dealing with troubleshooting issues and any distractions that may hijack the troubleshooting process should be subdued.
+The health management APIs in Netdata allows teams to eliminate unnecessary alerting during scheduled maintenance, testing, auto scaling events, and instance reboots.
+ 
+## Health Management API
+
+Netdata v1.12 and beyond provides a command API to control health checks and notifications at runtime. The feature is especially useful for maintenance periods, during which you receive meaningless alerts. From Netdata v1.16.0 and beyond, the configuration controlled via the API commands is [persisted across Netdata restarts](https://learn.netdata.cloud/docs/agent/web/api/health#persistence).
```
Comment (DShreve2) [inline blog/disable-alerts-health-management-api.md:15]: I think, this far ahead of the version, we can say (for the blog at least), just "Netdata provides a command API..."

Same with the next mention. Let's just say it exists without version specificity. 

You okay with that, @cakrit?

**Review by DShreve2 — APPROVED** [review]

File: blog/disable-alerts-health-management-api.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
+---
+slug: Mute Alerts 
+title: "How to mute alerts during maintenance windows or scheduled backups?"
+description: "Disable or Mute Alerts during maintenance windows"
```
Comment (DShreve2) [inline blog/disable-alerts-health-management-api.md:4]: We'll need to add a preview image for this, too. Just a line at line five that has:

image: [image url]

**Review by DShreve2 — APPROVED** [review]

File: blog/disable-alerts-health-management-api.md (hunk 1 above)
Comment (cakrit) [reply blog/disable-alerts-health-management-api.md]: Yes, that version is ancient.

File: blog/disable-alerts-health-management-api.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+The health management APIs in Netdata allows teams to eliminate unnecessary alerting during scheduled maintenance, testing, auto scaling events, and instance reboots.
+
+<!--truncate-->
+
+For all SREs, it is absolutely crucial to filter out expected events during maintenance windows and quickly pinpoint critical issues in your infrastructure. Every minute is crucial while dealing with troubleshooting issues and any distractions that may hijack the troubleshooting process should be subdued.
+The health management APIs in Netdata allows teams to eliminate unnecessary alerting during scheduled maintenance, testing, auto scaling events, and instance reboots.
+ 
+## Health Management API
+
+Netdata provides a command API to control health checks and notifications at runtime. The feature is especially useful for maintenance periods, during which you receive meaningless alerts. The configurations controlled via the API commands is [persisted across Netdata restarts](https://learn.netdata.cloud/docs/agent/web/api/health#persistence).
```
Comment (shyamvalsan) [inline blog/disable-alerts-health-management-api.md:19]:
```suggestion
Netdata provides a command API to control health checks and notifications at runtime. The feature is especially useful for maintenance periods, during which you receive meaningless alerts. The configurations controlled via the API commands are [persisted across Netdata restarts](https://learn.netdata.cloud/docs/agent/web/api/health#persistence).
```

File: blog/disable-alerts-health-management-api.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+```yaml
+[registry]
+   # netdata management api key file = /var/lib/netdata/netdata.api.key
+```
+ 
+You can access the API via GET requests, by adding the bearer token to an Authorization http header, like this:
+```bash
+curl "http://NODE:19999/api/v1/manage/health?cmd=RESET" -H "X-Auth-Token: Mytoken"
+```
+By default access to the health management API is only allowed from localhost. Accessing the API from anything else will return a 403 error with the message You are not allowed to access this resource. You can change permissions by editing the allow management from the variable in netdata.conf within the [web server access lists](https://learn.netdata.cloud/docs/agent/web/server#access-lists) section. See web server access lists for more information.
```
Comment (shyamvalsan) [inline blog/disable-alerts-health-management-api.md:35]:
```suggestion
By default, access to the health management API is restricted to localhost. Accessing the API from anywhere else will return a 403 error with the message "You are not allowed to access this resource". You can change permissions by editing the `allow management from` variable in netdata.conf within the [web server access lists](https://learn.netdata.cloud/docs/agent/web/server#access-lists) section. See web server access lists for more information.
```

File: blog/disable-alerts-health-management-api.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+If all you need is temporarily disable all health checks, then you issue the following before your maintenance period starts:
+```bash
+curl "http://NODE:19999/api/v1/manage/health?cmd=DISABLE ALL" -H "X-Auth-Token: Mytoken"
+```
+ 
+The effect of disabling health checks is that the alarm criteria are not evaluated at all and nothing is written in the alarm log. If you want the health checks to be running but to not receive any notifications during your maintenance period, you can instead use this:
+```bash
+curl "http://NODE:19999/api/v1/manage/health?cmd=SILENCE ALL" -H "X-Auth-Token: Mytoken"
+```
+Alarms may then still be raised and logged in Netdata, so you'll be able to see them via the UI.
```
Comment (shyamvalsan) [inline blog/disable-alerts-health-management-api.md:53]: Should we also talk about how the user can just turn off notifications from the UI? They can do this when a maintenance window starts and turn them back on after its done. 

Would be good to drop a screenshot of this as well - the new one that Alexandros demo'd today.

**Review by shyamvalsan — APPROVED** [review]

File: blog/disable-alerts-health-management-api.md (hunk 1 above)
Comment (sashwathn) [reply blog/disable-alerts-health-management-api.md]: This is fixed.

File: blog/disable-alerts-health-management-api.md (hunk 2 above)
Comment (sashwathn) [reply blog/disable-alerts-health-management-api.md:4]: I will put something up..

File: blog/disable-alerts-health-management-api.md (hunk 5 above)
Comment (sashwathn) [reply blog/disable-alerts-health-management-api.md:53]: Yes, I think I will add this. Although, the users will currently need to do both - from the Agent and from the Cloud.

File: blog/disable-alerts-health-management-api.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+## List silencers
+
+The command LIST was added in Netdata v1.16.0 and returns a JSON with the current status of the silencers.
+```bash
+curl "http://NODE:19999/api/v1/manage/health?cmd=LIST" -H "X-Auth-Token: Mytoken"
+```
+
+As an example, the following response shows that we have two silencers configured, one for an alarm called samplealarm and one for alarms with context random on host myhost
+
+```yaml
```
Comment (andrewm4894) [inline blog/disable-alerts-health-management-api.md:122]:
```suggestion
```json
```

File: blog/disable-alerts-health-management-api.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                       "context": "random",
+                       "hosts": "myhost"
+               }
+       ]
+}
+```
+
+The response below shows that we have disabled all health checks.
+
+```yaml
```
Comment (andrewm4894) [inline blog/disable-alerts-health-management-api.md:141]:
```suggestion
```json
```

File: blog/disable-alerts-health-management-api.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+json
```
Comment (andrewm4894) [inline blog/disable-alerts-health-management-api.md:142]:
```suggestion
```

File: blog/disable-alerts-health-management-api.md (hunk 5 above)
Comment (sashwathn) [reply blog/disable-alerts-health-management-api.md:63]: Done, but with the old snapshots still.. I will take a look at what Alexandros delivers (if not already in).

File: blog/disable-alerts-health-management-api.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+{
+       "all": true,
+       "type": "DISABLE",
+       "silencers": []
+}
+```
+ 
+ 
+
+**Note**: The system catalog table pg_attribute is usually the largest catalog table. If the pg_attribute table is significantly bloated, a VACUUM FULL operation on the table might require a significant amount of time and might need to be performed separately. The presence of both of these conditions indicate a significantly bloated pg_attribute table that might require a long VACUUM FULL time.
```
Comment (shyamvalsan) [inline blog/disable-alerts-health-management-api.md:157]: This **Note** is from another blog I guess - should be removed.

File: blog/2022-11-03-disable-alerts-health-management-api.md [hunk 10]
```diff
@@ -19,10 +19,9 @@ The health management APIs in Netdata allows teams to eliminate unnecessary aler
 
 At the beginning of any maintenance window, you can turn off all the notifications for a user at space or even at the war room level.
 
-<img width="700" alt="image1" src="https://user-images.githubusercontent.com/96257330/199587557-5495e465-ea7a-408b-8d80-1d7f7562712f.png">
+![image1](https://user-images.githubusercontent.com/96257330/199587557-5495e465-ea7a-408b-8d80-1d7f7562712f.png)
```
Comment (andrewm4894) [inline blog/2022-11-03-disable-alerts-health-management-api.md:22]: @sashwathn this was what was breaking it. need do images like this.

**Comment by andrewm4894** [comment]
@sashwathn filename also needs to be like `yyyy-mm-dd-slug-of-post.md`

**Comment by andrewm4894** [comment]
fixed and good to merge whenever

**Comment by sashwathn** [comment]
Thanks @andrewm4894 !

**Pull request merged**